pkg/trace/agent/agent.go

```diff
@@ -372,6 +365,22 @@ func (a *Agent) Process(p *api.Payload) {
 	}
 }
 
+// processedTrace creates a ProcessedTrace based on the provided chunk and root.
+// It makes a deep copy of the provided chunk to ensure that any subsequent changes
+// to the original chunk will not affect the TraceChunk of the ProcessedTrace.
+func processedTrace(p *api.Payload, chunk *pb.TraceChunk, root *pb.Span) traceutil.ProcessedTrace {
+	ptChunk := new(pb.TraceChunk)
+	*ptChunk = *chunk
+	return traceutil.ProcessedTrace{
+		TraceChunk:             ptChunk,
+		Root:                   root,
```

Comment:
We could probably copy `*root` as well to be extra careful.

Reply:
I think I'll do this in the follow-up PR that does a few more fixes. (e.g. around events)